pydantic/json_schema.py

```diff
@@ -958,17 +958,25 @@ def dict_schema(self, schema: core_schema.DictSchema) -> JsonSchemaValue:
         """
         json_schema: JsonSchemaValue = {'type': 'object'}
 
-        keys_schema = self.generate_inner(schema['keys_schema']).copy() if 'keys_schema' in schema else {}
+        keys_core_schema = schema.get('keys_schema')
+        self.generate_inner(keys_core_schema).copy() if keys_core_schema is not None else {}
```

Comment:
```suggestion
        keys_schema = self.generate_inner(schema['keys_schema']).copy() if 'keys_schema' in schema else {}
```

Walking back my original suggestion, just because we don't need to use the `keys_core_schema` anywhere